modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminDisplayContext.java

```diff
@@ -802,11 +803,17 @@ private Hits _getHits(SearchContainer<RepositoryEntry> searchContainer)
 		SearchContext searchContext = SearchContextFactory.getInstance(
 			_httpServletRequest);
 
-		searchContext.setAttribute("paginationType", "regular");
-
 		long searchRepositoryId = ParamUtil.getLong(
 			_httpServletRequest, "searchRepositoryId");
 
+		Group group = GroupLocalServiceUtil.fetchGroup(searchRepositoryId);
+
+		if (group != null) {
+			searchContext.setGroupIds(new long[] {searchRepositoryId});
+		}
+
+		searchContext.setAttribute("paginationType", "regular");
+
```

Comment:
@thienquach87 You should check that the user has VIEW permissions over the group. Otherwise malicious users could forge URLs to search content in groups they have no access to:
```
if ((group != null) && GroupPermissionUtil.contains(themeDisplay.getPermissionChecker(), group, ActionKeys.VIEW)) {
    ...
}
```

Reply:
@adolfopa I added your condition